src/network/websockets/server/index.ts

```diff
@@ -44,6 +44,22 @@ export function runWebsocketServer(params: Params) {
 
     const handlers: { [action: string]: (msg: any) => void } = {
       ping: () => send({ action: "pong" }),
+      auth: msg => {
+        const { token } = msg;
+        onAuthenticate({ token })
+          .then(function(result) {
+            send({
+              action: "authResult",
+              result: result
+            });
+          })
+          .catch(function() {
+            send({
+              action: "authResult",
+              result: "internalError"
```

Comment:
perhaps we should return and actual Error? or something equivalent? do we have any president on the project for returning user facing errors?

```js
{ type: "error, message: "internal error" }
```

Reply:
No, we’re not sending any details to the client at the moment. I would rather handle any “error” as a special case scenario and design the workflow for it. The internal error is a special case for a panic scenario.